library/lib/serve.js

```diff
@@ -138,26 +131,69 @@ function possibleDirectories(path) {
   return possibleDirectories
 }
 
-function serveIndexWithRouting(req, res, file) {
+function serveIndexWithRouting(file, { req, res, next }) {
   const envRequire = isProduction ? require : require('import-fresh')
+
   const routeTemplate = envRequire(file)
 
-  const routes = routeTemplate.routes
   const location = parsePath(req.url)
 
-  return Promise.resolve(routes)
-    .then(routes => (routes && routes.match(location, req)) || { status: 200, data: null })
-    .then(data => {
-      if (!routes || !routes.resolveIndex) return [data, routeTemplate]
-
-      const indexLocation = routes.resolveIndex(location, req)
-      if (!indexLocation) return [data, routeTemplate]
-
-      const indexPath = resolve(target, publicPathDir, indexLocation, indexWithRouting)
-      return [data, envRequire(indexPath)]
-    })
-    .then(([{ status, headers, data }, template]) =>
-      Promise.resolve(template({ location, data })).then(html => [status, headers, html])
-    )
-    .then(([ status, headers, html ]) => res.status(status).set(headers).send(html))
+  const [dataOrPromise, template] = getDataAndRouteTemplate(routeTemplate, { location, req })
```

Comment:
Ik zou hier geen object aanmaken
```suggestion
  const [dataOrPromise, template] = getDataAndRouteTemplate(routeTemplate, req, location)
```